src/main/java/ru/practicum/shareit/request/service/ItemRequestServiceImpl.java

```diff
@@ -0,0 +1,80 @@
+import ru.practicum.shareit.request.mapper.ItemRequestMapper;
+import ru.practicum.shareit.request.model.ItemRequest;
+import ru.practicum.shareit.request.storage.ItemRequestRepository;
+import ru.practicum.shareit.user.model.User;
+import ru.practicum.shareit.user.storage.UserRepository;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor(onConstructor_ = @Autowired)
+public class ItemRequestServiceImpl implements ItemRequestService {
+    private final ItemRepository itemRepository;
+    private final ItemRequestRepository itemRequestRepository;
+    private final UserRepository userRepository;
+
+    @Override
+    public ItemRequestDto create(long userId, ItemRequestDto itemRequestDto) {
+        User user = userRepository.findById(userId).orElseThrow(() -> {
+            throw new NotFoundException("Пользователь не найден");
+        });
+        itemRequestDto.setCreated(LocalDateTime.now());
+        ItemRequest itemRequest = itemRequestRepository.save(ItemRequestMapper.toItemRequest(itemRequestDto, user));
+        return ItemRequestMapper.toItemRequestDto(itemRequest);
+    }
+
+    private List<ItemRequestDto> addItemsToRequests(List<ItemRequestDto> requestsDtoList) {
+        List<Long> requestsIds = new ArrayList<>();
+        requestsDtoList.forEach(itemRequestDto -> requestsIds.add(itemRequestDto.getId()));
+        List<ItemDto> items = itemRepository.findByRequestsIds(requestsIds).stream().map(ItemMapper::toItemDto).collect(Collectors.toList());
+        Map<Long, ItemRequestDto> requestsMap = requestsDtoList.stream().collect(Collectors.toMap(ItemRequestDto::getId, film -> film, (a, b) -> b));
+        items.forEach(itemDto -> requestsMap.get(itemDto.getRequestId()).getItems().add(itemDto));
+        return new ArrayList<>(requestsMap.values());
+    }
+
+    @Override
+    public List<ItemRequestDto> getRequestsInformation(long userId) {
+        User user = userRepository.findById(userId).orElseThrow(() -> {
+            throw new NotFoundException("Пользователь не найден");
+        });
+        List<ItemRequestDto> requestsList = itemRequestRepository.findAllByRequestorId(userId).stream().map(ItemRequestMapper::toItemRequestDto).collect(Collectors.toList());
+        return addItemsToRequests(requestsList);
+    }
+
+    @Override
+    public ItemRequestDto getRequestInformation(long userId, long request) {
+        User user = userRepository.findById(userId).orElseThrow(() -> {
+            throw new NotFoundException("Пользователь не найден");
+        });
+        ItemRequest itemRequest = itemRequestRepository.findById(request).orElseThrow(() -> {
+            throw new NotFoundException("Запрос не найден");
+        });
+        List<ItemDto> itemsList = itemRepository.findByItemRequestId(request).stream().map(ItemMapper::toItemDto).collect(Collectors.toList());
```

Comment:
Для лучшей читаемости стримы принято бить по строкам:
```suggestion
        List<ItemDto> itemsList = itemRepository.findByItemRequestId(request).stream()
            .map(ItemMapper::toItemDto)
            .collect(Collectors.toList());
```

Поправь, пожалуйста, во всех новых методах